### ISSUE TITLE
Does anyone have any large/huge TreeSheets documents they could share for import testing?

### ISSUE THREAD

**Issue by merriman-xyz** [aardappel/treesheets#199]
I'm working on webbish TreeSheet-type app (an Electron app at the moment) and was wondering if you or anyone else has large/huge TreeSheet documents they can share for testing?

I am able to import the XML version of the `tutorial.cts`:

![Screenshot from 2022-01-31 20-53-15](https://user-images.githubusercontent.com/67341535/151902051-7696decb-8537-47c3-9fbc-02eac2515a8f.png)

And the XML version of the `todo_calendar.cts` file:

![Screenshot from 2022-01-31 20-58-46](https://user-images.githubusercontent.com/67341535/151902417-a3ac85e5-48bd-476f-ba7a-c6063d4e8efe.png)

And wanted to try some larger, non-trivial TreeSheet files.

**Comment by aardappel** [comment]
Wow, this could be very useful for lots of people, thanks for working on it!

I have some mega docs, sadly they are all rather personal.

Couldn't you simply take the tutorial, and paste a copy of each into an NxM grid for testing?

**Comment by merriman-xyz** [comment]
> Couldn't you simply take the tutorial, and paste a copy of each into an NxM grid for testing?

Good idea! Wildly copied the tutorial a few times and ended up at about ~3k nodes/cells.

https://user-images.githubusercontent.com/67341535/151906711-833d9628-5970-4559-828e-0aabf0e39dab.mp4

Will try mixing the tutorial and example TreeSheets together to see what happens. 

I am also interested in "real" user type data as I have only been running tests on this sort of mock data. 

> I have some mega docs, sadly they are all rather personal.

Understand!

Do you have any sense of how many cells they are? Just getting a sense of how many cells a *large* or *huge* grid has would still be of great help. 

Thanks for responding, it is much appreciated!

**Comment by aardappel** [comment]
> Do you have any sense of how many cells they have?

Large files I have are about a megabyte each .cts files, contain no images, and relatively low amount of text per cell..

**Comment by ojg78** [comment]
Hi, just curious if development went anywhere on this?

**Comment by noyannus** [comment]
Late, but maybe still useful:

For a stress test with deeply nested cells, copy-paste some sheet like the tutorial into a (as deeply nested as possible) cell of itself: If you want more layers, copy the whole top grid into one of the now deepest nested cells of itself. (With the tutorial as example you get tutorial nested in tutorial nested in tutorial nested in ...) Repeat ad lib, doubling each time the depth of the cell-in-cell nesting.

With the same principle double the sheets width -- copy itself into its [left|right]most border.

Ditto with height and [top|bottom].

Combine all three, until you run out of RAM. Close without saving, then repeat the above with >=1 steps less, so you will have a little RAM to keep working. Or you monitor RAM while doubling the sheet every time. Note that omitting 1 step already halves its RAM hunger.

**Comment by merriman-xyz** [comment]
> Hi, just curious if development went anywhere on this?

Hey this is 3ish years late, but have an alpha up here https://merriman-xyz.github.io/tanji-systems/ingrid/

Any feedback welcome here https://github.com/merriman-xyz/Ingrid-feedback

**Comment by ib-bsb-br** [comment]
> > Hi, just curious if development went anywhere on this?
> 
> Hey this is 3ish years late, but have an alpha up here [merriman-xyz.github.io/tanji-systems/ingrid](https://merriman-xyz.github.io/tanji-systems/ingrid/)
> 
> Any feedback welcome here [merriman-xyz/Ingrid-feedback](https://github.com/merriman-xyz/Ingrid-feedback?rgh-link-date=2026-06-18T17%3A08%3A38.000Z)

no fucking way, so awesome!

**Comment by aardappel** [comment]
@merriman-xyz wow, very cool.. seems very functional already :)

My initial comments:
* Having a double-line between all the cells makes it hard to visually follow the structure.. I'd find a way to make that a single line.
* I couldn't find a "paste" operation (for multiple cells, not just text pasting).
* I'd find a way to not show the extra buttons under Math, Code etc all the time, maybe only when selected.. looks very cluttered.
